### PR TITLE
executor, expression: support Chunk in ProjectionExec

### PR DESCRIPTION
**Pull request by zz-jason**
To simplify code, I haven't support vectorized expression evaluation here, to support it, we have to check whether there exists some special expressions, I'll do it in the near future.

Here is an example that we have to evaluate expressions row by row:

```sql
create table t(a bigint);
insert into t values(1), (2), (3);
select @a, @a := a+1 from t;
```

Main changes:
1. Let `ProjectionExec` support chunk and add implement function `NextChunk`.
2. Implemented `EvalExprsToChunk` in package **expression** to evaluate a list of expression based on an input Chunk and append all the result to an output Chunk. We can further improve it to evaluate output Chunk column by column, ie. support vectorized expression evaluation.
3. Add a new function named `newChunk` to interface `executor`

### CODE REVIEW

File: expression/expression.go [hunk 1]
```diff
@@ -85,6 +87,102 @@ type Expression interface {
 	ExplainInfo() string
 }
 
+// EvalExprsToChunk evaluates a list of expressions and append their results to "output" Chunk.
+// TODO: Determine whether some of the expressions have internal depencies and implement a vectorized execution method if not.
+func EvalExprsToChunk(ctx context.Context, exprs []Expression, input, output *chunk.Chunk) error {
+	sc := ctx.GetSessionVars().StmtCtx
+	for rowID, length := 0, input.NumRows(); rowID < length; rowID++ {
+		for colID, expr := range exprs {
```
Comment (coocood) [inline expression/expression.go]: It would be more cache friendly If we move `exprs` range to outer for loop.

File: expression/expression.go [hunk 2]
```diff
@@ -85,6 +87,102 @@ type Expression interface {
...
+			if err != nil {
+				return errors.Trace(err)
+			}
+		}
+	}
+	return nil
+}
+
+func evalOneCell(sc *variable.StatementContext, expr Expression, input, output *chunk.Chunk, rowID, colID int) error {
+	switch fieldType, evalType := expr.GetType(), expr.GetType().EvalType(); evalType {
```
Comment (coocood) [inline expression/expression.go]: The `EvalType` is constant for every row, so we can loop all the rows under a certain eval type case.

**Comment by coocood** [comment]
LGTM

File: expression/expression.go [hunk 3]
```diff
@@ -85,6 +87,102 @@ type Expression interface {
...
+	case types.ETInt:
+		res, isNull, err := expr.EvalInt(input.GetRow(rowID), sc)
+		if err != nil {
+			return errors.Trace(err)
+		} else if isNull {
```
[the remark this answers is not in the document]
Comment (zz-jason) [reply expression/expression.go]: ok, I'll do it in the next pr

**Pull request merged**